python/ray/serve/tests/test_handle.py

```diff
@@ -35,6 +35,57 @@ def __call__(self):
     assert requests.get("http://127.0.0.1:8000/endpoint2").text == "hello"
 
 
+def test_handle_http_args(serve_instance):
+    serve.init()
+
+    class Endpoint:
+        def __init__(self):
+            self.handle = serve.get_handle("endpoint1", missing_ok=True)
+
+        def __call__(self, request):
+            return {
+                "args": dict(request.args),
+                "headers": dict(request.headers),
+                "method": request.method,
+                "json": request.json
+            }
+
+    serve.create_backend("backend", Endpoint)
+    serve.create_endpoint(
+        "endpoint", backend="backend", route="/endpoint", methods=["POST"])
+
+    grond_truth = {
+        "args": {
+            "arg1": "1",
+            "arg2": "2"
+        },
+        "headers": {
+            "X-Custom-Header": "value"
+        },
+        "method": "POST",
+        "json": {
+            "json_key": "json_val"
+        }
+    }
+
+    resp_web = requests.post(
+        "http://127.0.0.1:8000/endpoint?arg1=1&arg2=2",
+        headers=grond_truth["headers"],
+        json=grond_truth["json"]).json()
+
+    handle = serve.get_handle("endpoint")
+    resp_handle = ray.get(
+        handle.options(
+            http_method=grond_truth["method"],
+            http_headers=grond_truth["headers"]).remote(
+                grond_truth["json"], **grond_truth["args"]))
```

Comment:
nit
```suggestion
                ground_truth["json"], **grond_truth["args"]))
```
and elsewhere